### PR TITLE
Update to Rust Nightly 2023-06-02 (stable 1.70.0)

### PR DESCRIPTION
**Pull request by W95Psp**
Updating was not too bothering, so that's a bit a relief

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
The nightly version in the readme needs updating (or just point to the toolchain toml so we don't need to keep them in sync).

The diagnostics crate needs an `extern crate rustc_driver;` to fix the crate (I pushed this change on your branch here).

**Comment by W95Psp** [comment]
There are some issues with the macro information recovering phase :(
Rustc now wraps surface ASTs in a `Steal` structure, and we're facing again stealing issues (but now only for the surface AST)

**Review by franziskuskiefer — COMMENTED** [review]
The changes itself look good to me but you are re-introducing the stealing bug.

File: frontend/exporter/src/all.rs [hunk 1]
```diff
@@ -893,95 +901,79 @@ pub struct Block {
...
+        Some(id) => s.tcx().param_env(id),
+        None => rustc_middle::ty::ParamEnv::empty(),
     }
-    // let impl_source = selection_ctxt.select(&obligation).unwrap().unwrap();
-    // let nested = impl_source.clone().nested_obligations();
 }
 
-impl<'tcx, S: BaseState<'tcx>> SInto<S, ProjectionTy> for rustc_middle::ty::ProjectionTy<'tcx> {
-    fn sinto(&self, s: &S) -> ProjectionTy {
+// fn resolve_trait<'tcx, S: BaseState<'tcx>>(trait_ref: rustc_middle::ty::TraitRef<'tcx>, s: &S) {
```
[the remark this answers is not in the document]
Comment (W95Psp) [reply frontend/exporter/src/all.rs]: I left the function `resolve_trait` there finally, but commented it out, and renamed it `_resolve_trait`.
For now we don't use it, but we will need it soon for impl resolution (#20)

**Review by franziskuskiefer — APPROVED** [review]
:shipit: 
bors r+

**Pull request merged**